src/mpid/ch4/netmod/ofi/ofi_nic.h

```diff
@@ -174,71 +157,27 @@ static int setup_multi_nic(void)
-    int close_nic_index = 0;
-    int num_close_nics = MPIDI_OFI_global.num_close_nics;
-    MPIR_Assert(num_close_nics != 0);
+    /* Because we cannot communicate with the other local processes to avoid collisions with the
+     * same NICs, just shift NICs that have multiple close NICs around according to their local
+     * rank. This will likely give the same result as long as processes have been bound properly. */
+    MPIDI_OFI_nic_info_t *old_nics = MPL_malloc(sizeof(MPIDI_OFI_nic_info_t) *
+                                                MPIDI_OFI_global.num_nics, MPL_MEM_ADDRESS);
+    memcpy(old_nics, nics, sizeof(MPIDI_OFI_nic_info_t) * MPIDI_OFI_global.num_nics);
 
-    for (int i = 0; i < local_size; ++i) {
-        if (have_same_nic_parents(close_nic_parents, i, local_rank)) {
-            if (i == local_rank) {
-                nic_parent_rank = nic_parent_rank_count;
-                nics[close_nic_index].prefer = current_prefer++;
-            }
-            nics[close_nic_index].count++;
-            for (int j = 0; j < num_close_nics; ++j)
-                nics[j].num_close_ranks++;
-            close_nic_index = (close_nic_index + 1) % num_close_nics;
-            nic_parent_rank_count++;
-        }
+    int old_idx = 0;
+    if (MPIDI_OFI_global.num_close_nics != 0) {
+        old_idx = local_rank % MPIDI_OFI_global.num_close_nics;
     }
 
-    /* Determine ordering of the close NICs based on count. */
-    /* After this qsort(), the NICs will be properly sorted
-     * except for the non-preferred close NICs. These need to
-     * be further sorted so their distribution is even across
-     * the NIC-local ranks. */
-    qsort(nics, num_close_nics, sizeof(nics[0]), compare_nics);
+    for (int new_idx = 0; new_idx < MPIDI_OFI_global.num_close_nics; new_idx++) {
+        if (old_idx >= MPIDI_OFI_global.num_close_nics)
+            old_idx = 0;
 
-    /* Go through the remaining NICs after picking the first preferred NIC and
-     * order based on count (the number of process's using the NIC as the
-     * preferred NIC) and the process's NIC parent rank. This will place less
-     * used NICs earlier in the NIC list while also spreading them
-     * evenly across processes. */
-    int start_index = 1;
-    while (start_index < num_close_nics) {
-        int end_index = start_index;
-        while (end_index < num_close_nics && nics[end_index].count == nics[start_index].count)
-            end_index++;
-        /* NICs from start_index to end_index will have the same count */
-        int num_count = end_index - start_index;
-        int offset = nic_parent_rank % num_count;
-        for (int i = 0; i < num_count; ++i) {
-            int real_index = start_index + ((offset + i) % num_count);
-            nics[real_index].prefer = current_prefer++;
-        }
-        start_index = end_index;
+        if (new_idx != old_idx)
+            memcpy(&nics[new_idx], &old_nics[old_idx], sizeof(MPIDI_OFI_nic_info_t));
```

Comment:
If the original `old_idx` is 0, you don't need rotate at all, so it probably should be checked after line 169 and skip this rotation altogether. If the original `old_idx` is not `0`, then every entry need be copied (since it is a rotation), you don't need this check.

Reply:
Yep. Put the whole thing in an `if` block to avoid all of this if not needed.